docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+
+However, this feature might not be important for the MVP.
+
+### How do we extend the lightning channel for DLCs?
+
+DLCs are not supported by the lightning channels outof the box and a custom extension needs to be implemented. Downwards compatibility should be granted though.
+
+ - **Option 1: *Virtual Channels***: ..
+ - **Option 2: ...**:
+
+## State management
+
+How are state changes managed in 10101.
+
+ - **Option 1: *Events***: State changes are reflected with immutable events allowing to recreate a state by applying all events in the correct order. While this architectural pattern introduces some complexities it allows for a much more flexible, async processing. In itchysats an event driven architecture has been implemented for parts of the solution.
+ - **Option 2: *API***: State changes are promoted through synchronous api calls with a blocking wait to check if the call has been processed successfully. This is a simple paradigm an less complex.  
+
+*Note, these options are not exhaustive nor very general for state management. Also the target architecture could make use of a combination of both approaches. Further elaboration and discussion is required here.* 
+
+## Coordinator API
+
+The coordinator will have to expose an API along the p2p API for the lightning (including dlc) communication e.g. order book. **Assumption**: 10101 will not exclusively talk via the p2p protocol with the app. As DLCs should get integrated into upstream rust-lightning a more generalised approach would be beneficiary.  
+
+- **Option 1 *Restfull API***: Simple API to expose CRUD operations through an HTTP. Broadly used and well understood.
+- **Option 2 *Websocket***: Only one API, exposing all sorts of data the client can subscribe to.
+
+A combination of both options seem to be reasonable as the approach in itchysats with only web sockets seemed artificial and introduced unneeded complexities. However using websockets to push updates to subscribed data objects (instead of polling) seems to be a good practice for multiple reasons (bandwidth, freshness, etc.).
+
+*We also shortly discussed graphql, but probably figured it to be not beneficiary (as too powerful and complex) for our use case.*
+  
+### RPC vs REST: 
+
+It seems RPC is broadly used nowadays. We might want to consider evaluating RPC for our Coordinator APIs.
+
+## Security
+
+One of our most important features is Security. We must not only think of it from a point of view that the protocol is secure. Equally we need to look at the way we derive and manage crypto materials. 
+
+### How to secure the cryptographic key on the phone?
+
+- **Option 1 *HSM***: iPhone and Android support hardware security modules on the phone. For iPhone its Secure Element - We could use such module (isolated from the os) to securily perfom cryptographic algorithms without ever exposing the private key. *Note, its unclear what that would mean for exporting the key if required* 
```

Comment:
Interesting idea. Is this then portable? E.g. can you create a back-up to recover your funds in case you lose your phone?

Reply:
> *Note, its unclear what that would mean for exporting the key if required* 

I haven't looked into that yet. But I think it's worthwhile to look into that in more detail.

Reply:
https://www.thecodeside.com/2020/09/14/android-auto-backup-keystore-encryption-broken-heart-love-story/

This article seems to suggest that it is not possible to back up data encrypted using a key stored in an Android HSM

Reply:
Then let's rule it out for now.